source/routing/raptor_solution_reader.cpp

```diff
@@ -318,16 +345,30 @@ struct RaptorSolutionReader {
                 && lhs.transfer_dur <= rhs.transfer_dur;
         }
     };
-    typedef std::map<JpIdx, ParetoFront<Transfer, DomTr>> Transfers;
+    template<typename T> struct Cache {
```

Comment:
Deserves a comment at least
Maybe a bench because we don't want to add code complexity here if the gains are negligible.